### ISSUE TITLE
The SP metadata generated by 1.4.0 can not be imported into ADFS when configuring relying party trust

### ISSUE THREAD

**Issue by ctosgh** [onelogin/python3-saml#94]
With the latest 1.4.0 release, the SP metadata file generated by library can not be imported into ADFS when configuring Relying Party Trust. The error message from ADFS is "Cannot resolve the '' URI in the signature to compute the digest".

Looking at the SP metadata, the "Signature" node is moved out of node "SPSSODescriptor" and becomes a sibling of node "SPSSODescriptor".  Looks like the 1.4.0 version introduce this in purpose.

Looks like two additional tasks need to be done if the position change is introduced.
1 Add "ID" attribute for node "SPSSODescriptor"
2 Add the attribute "URI" for node "Reference" and value of "URI" is "pointing" to the node "SPSSODescriptor"

Correct me if I am wrong.

**Comment by ctosgh** [comment]
Good to know we have the same understanding. Is the fix will be included in the next release?

**Comment by pitbulk** [comment]
@ctosgh can you test if the code on master fixed your issue?

**Comment by ctosgh** [comment]
Hi, Sixto

I tried with the master branch and it work fine! The reference node has the
URI attribute and it can be imported into adfs automatically~


Thanks very much!


Thanks,
Jacky

2018-04-24 9:19 GMT-07:00 Jacky Sun <ctosgh@gmail.com>:

**Comment by pitbulk** [comment]
The ID is generated randomly and is used to identify a SAML element.
The current code of java-saml will generate it.

**Comment by ankur11** [comment]
Yes, but when I generate SP metadata, that ID is not shown in EntityDescriptor nor in Reference URI. I have to integrate it with ADFS and it needs that Reference URI.
Do I have to manually add that after the metadata has been generated ?
I am using Python3-saml toolkit.
I am using below command to verify the xml generated
xmlsec1 --verify --X509-skip-strict-checks --id-attr:ID EntityDescriptor --id-attr:URI Reference --trusted-pem certificates/certificate.crt test_xml_metadata.xml
